tests/test_resolver.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The Foundry Visionmongers Ltd
+
+# pylint: disable=no-self-use
+# pylint: disable=invalid-name,redefined-outer-name
+# pylint: disable=missing-class-docstring,missing-function-docstring
+
+import pytest
+import os
+
+# This environment var must be set before the usd imports.
+os.environ["TF_DEBUG"] = "OAIO_RESOLVER"
+from pxr import Plug, Usd
+
+
+@pytest.fixture(autouse=True)
+def register_plugin():
+    # plugInfo.json file for the example resolver should be installed
+    # in the resources folder under the test directory.
+    Plug.Registry().RegisterPlugins(os.path.join(os.getcwd(), "resources"))
+
+
+def test_open_stage_and_logging(capfd):
+    stage = Usd.Stage.Open("resources/empty_shot.usda")
+    captured = capfd.readouterr()
+    assert "_CreateIdentifier" in captured.out
```

Comment:
[consider][minor] Perhaps a little out of scope for this PR, and less relevant once we integrate OpenAssetIO, but I think there would be some value in asserting the call order and full text.  

We can filter for lines containing the common prefix (currently "OAIO Resolver" though see Tom's comment), then construct an expected list of calls in order.

Reply:
I'm concerned that asserting the full text would result in a lot of awkward string wrangling to match the formatting, there's a large amount of text output when you include all the output, including lots of absolute paths that would have to retargetted in the test code.

The thing to do perhaps is to assert the order, but continue to omit the arguments.

Reply:
[eecb4b9](https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/3/commits/eecb4b910929e1cdd75389144969733a071b06dd)

Reply:
Good call - dont forget this specific test is throwaway as we'll be removing it as soon as we add actual functionality!